### PR TITLE
INTMDB-713: [CFN-PI] Enable golangci for go tests files

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

- Enables golangci for go tests files
- Deletes cluster-legacy: this resource is not used (we have [cluster](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/cfn-resources/cluster))
- Makes `CreateMongoDBClient` private as it is not longer used by Resources but only in util.

_Jira ticket:_ [INTMDB-713](https://jira.mongodb.org/browse/INTMDB-713)

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/.golangci.yml [hunk 1]
```diff
@@ -130,17 +130,6 @@ issues:
...
-    - path: _test\.go
-      linters:
-        - testpackage
-      # part of the golangci govet package is picking up things that go vet doesn't. Seems flaky, shutting that specific error off
-
-    # Exclude some staticcheck messages
-    - linters:
-        - staticcheck
-      text: "SA1019:" # d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK
-
```
Comment (andreaangiolillo) [inline cfn-resources/.golangci.yml:143]: This is the main change.

**Review by gssbzn — APPROVED** [review]
I see a lot of deletions which I'm going to assume it's safe to do, LGTM and thanks fro the quick follow up

**Comment by andreaangiolillo** [comment]
@Zuhairahmed Could you confirm that `cluster-legacy` can removed? I see that the resource was not updated to use the Profile and that we have [cluster](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/cfn-resources/cluster)). Thanks!

**Comment by andreaangiolillo** [comment]
Checked with Zuhair and we can delete it

**Pull request merged**